tests/test_quickbooks_discovery.py

```diff
@@ -146,3 +147,15 @@ def test_run(self):
                          and item.get("breadcrumb", ["properties", None])[1]
                          not in actual_automatic_fields}),
                     msg="Not all non key properties are set to available in metadata")
+
+                # verify custom field schema
+                if stream in self.custom_command_streams:
+                    actual_custom_field_keys = list(schema["properties"]
```

Comment:
We should be testing this in metadata not annotated-schema. The menagerie method call is misleadingly named. But annotated-schema is no longer used by our front-end, only metadata is. You should be able to make the same assertion just change the way the actual value is accessed.

Reply:
@kspeer825 This test is validating nested fields present under the `CustomField` field.
However, as metadata only contains first-level fields, can you please help us on how to retrieve nested fields from metadata?

Reply:
I am not sure this is possible. The metadata is what drives table and field selection in our ui. And nested field selection is not a feature that I am aware of, so I think this would require a change to singer as well as our frontend. I will confirm this with a developer and get back to you.

Reply:
I confirmed this with the devs. I think in this test discovering the top level key from metadata is sufficient. If you need to confirm that the subfields are picked up by the tap, you would need a test that performs a sync and you could pull the actual field values as well as the json schema from the messages sent to the target.

Reply:
@kspeer825, Updated test, and accessed top-level field value from metadata. The subfields validation from sync is already covered as part of the `test_quickbooks_sync_all.py' test.